receiver.go

```diff
@@ -609,7 +628,9 @@ func (r *Receiver) mux() {
 			if r.l.doneErr != nil {
 				return
 			}
-
+		case count := <-r.settlementCount:
```

Comment:
This channel seems like a potential bottleneck. 

Can it just be an int64, along with the `atomic.Add*` and `atomic.Load` functions?

Reply:
I thought about that.  My concern is that, if not done properly, the link credit will become out of sync.  While I'm not against a more complicated solution, we should measure both to ensure that there's an actual problem.

Reply:
I did a little benchmarking and there is somewhat of an appreciable difference in perf.

***without this change (latest in main)***
```
msgCount: 1000, credit: 1, average duration: 70.205µs
msgCount: 1000, credit: 10, average duration: 20.909µs
msgCount: 10000, credit: 20, average duration: 15.803µs
msgCount: 20000, credit: 100, average duration: 11.308µs
```

***with this change***
```
msgCount: 1000, credit: 1, average duration: 195.753µs
msgCount: 1000, credit: 10, average duration: 65.399µs
msgCount: 10000, credit: 20, average duration: 59.568µs
msgCount: 20000, credit: 100, average duration: 40.87µs
```
So it's roughly 3x slower.

Let me see if I can improve on that.

Reply:
Switching to a mutex yeilds much better perf and is easy to reason about for correctness.
```
msgCount: 1000, credit: 1, average duration: 10.747µs
msgCount: 1000, credit: 10, average duration: 10.568µs
msgCount: 10000, credit: 20, average duration: 5.369µs
msgCount: 20000, credit: 100, average duration: 3.409µs
```

Reply:
This does highlight that we need some benchmark tests to measure various APIs.